File: tests/regex_test.py
#!/usr/bin/env python
from tests.parser_test_helper import *


class LoopTest(ParserBaseTest):
	def test_match(self):
		self.assert_that("'beep' ~ r'ee'")

	def test_match2(self):
		self.assert_that("'be4ep' ~ regex '\d' == 4")

	def test_match3(self):
		# skip()
		self.assert_that("'be4ep' ~ '\d' == 4")

File: angle.js/test/regex.test.js
#!/usr/bin/env python;



class RegExpTest extends (ParserBaseTest) {
	test_match(){
		self.assert_that(`'beep' ~ r'ee'`);
	}
	test_match2(){
		self.assert_that(`'be4ep' ~ regex '\d' == 4`);
	}
	test_match3(){
		// skip()
		self.assert_that(`'be4ep' ~ '\d' == 4`);
	}

	test_match4(){
		// skip()
		self.assert_that(`'be4ep' ~ /\d/ == 4`);
	}
}

register(RegExpTest, module)